### PR TITLE
Serkan - Tic-Tac-Toe Finished

### PR DESCRIPTION
**Pull request by serkanbal**

### CODE REVIEW

File: starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java [hunk 1]
```diff
@@ -1,13 +1,234 @@
...
+            win = true;
+        } else {
+            win = false;
+        }
+    }
 
-public class GameActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
```
Comment (charlesdrews) [inline starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java:111]: Best practice is usually to put `onCreate()` and other overridden methods from Activity at the top, right after your member variables, then put your helper methods below.

File: starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java [hunk 2]
```diff
@@ -1,13 +1,234 @@
...
+                        mTextView1.setText(mLetter);
+                        timer.start();
+                    }
+                    break;
+                case R.id.textView2:
+                    if (mTextView2.getText() == "") {
+                        mTurn++;
+                        whoseTurn();
+                        mTextView2.setText(mLetter);
+                        timer.start();
```
Comment (charlesdrews) [inline starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java:172]: You're repeating the same logic a bunch of times - I suggest moving this to another helper method and passing each TextView to that method when they get clicked.

**Review by charlesdrews — COMMENTED** [review]
👍 homework complete - great job!

File: starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java (hunk 2 above)
Comment (serkanbal) [reply starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java:172]: Hey Charlie,
To be honest; I also thought of that; but for every case; I had to put in a different TextView object and I couldn't quite figure out how. Can you write a simple example method for me that would take into account the changing TextViews?
Thank you.

Comment (charlesdrews) [reply starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java:172]: Great question! How about this:

``` java
public void handleTextViewSelection(TextView selectedTextView) {
    if (selectedTextView.getText().toString().isEmpty()) {
        mTurn++;
        whoseTurn();
        selectedTextView.setText(mLetter);
        timer.start();
    }
}
```

Then your `switch` would look something like:

``` java
switch (view.getId()) {
    case R.id.textView1:
    case R.id.textView2:
    case R.id.textView3:
    case R.id.textView4:
    ...
        handleSelectedText( (TextView) view ); // cast view to type TextView
        break;
}
```

Comment (serkanbal) [reply starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java:172]: Thank you! I will try to insert this and give it a try.

**Pull request closed without merging**